### PR TITLE
Feature/platooning strategic ros2

### PR DESCRIPTION
**Pull request by msmcconnell**
# PR Details
## Description
This PR ports the platooning strategic ihp plugin to ROS2 as well as fixes a number of bugs associated with recent ports. 
Changes:
* platoon_strategic_ihp ported to ROS2
* platoon_strategic removed as it has been fully replaced by platoon_strategic_ihp 
* Updates some platooning unit tests to compile and pass. They had been very out of date during ihp2 testing. There are still 7 failing tests which need to be updated by someone with more knowledge of the state machine. But these tests are also failing in ROS1. 
* ui.launch.py corrected to use the launch file from rosbrige_websocket. It was previously trying to load a python node as a component which is impossible. 
* Removes the UI parameters file which didn't have a clear path to loading and has never been modified in carma2 or 3 so defaults should be plenty sufficient. 
* Corrects a bug in the stop_and_wait_plugin where the wrong library was specified as the component making the class unloadable
* Corrects a bug in route_following_plugin where a decimal parameter was marked as an integer

This PR should be merged simultaneously with
https://github.com/usdot-fhwa-stol/carma-config/pull/207
and
https://github.com/usdot-fhwa-stol/carma-web-ui/pull/148

## Related Issue
Resolves https://github.com/usdot-fhwa-stol/carma-platform/issues/1895
Resolves https://github.com/usdot-fhwa-stol/carma-platform/issues/1896
Resolves https://github.com/usdot-fhwa-stol/carma-platform/issues/1897
Resolves https://github.com/usdot-fhwa-stol/carma-platform/issues/1898
Resolves https://github.com/usdot-fhwa-stol/carma-platform/issues/1899

## Motivation and Context
Functional develop builds and ROS2 ports

## How Has This Been Tested?
Local launch testing and launch testing on pacifica. Every platooning topic has been verified along with UI connection. I will test platooning execution one the full platooning stack has been ported. 

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: platooning_strategic_IHP/test/main_test.cpp [hunk 1]
```diff
@@ -1,27 +1,55 @@
...
-#include <ros/ros.h>
+#include <memory>
+#include <chrono>
+#include <thread>
+#include <future>
+
+#include "platoon_strategic_ihp/platoon_strategic_plugin_node_ihp.h"
+
+
+// TODO for USER: Implement a real test using GTest
```
[the remark this answers is not in the document]
Comment (msmcconnell) [reply platooning_strategic_IHP/test/main_test.cpp]: done

File: carma/launch/ui.launch.py [hunk 2]
```diff
@@ -38,70 +38,55 @@ def generate_launch_description():
...
+            SetRemap("get_drivers_with_capabilities", [EnvironmentVariable('CARMA_INTR_NS', default_value=''), "/get_drivers_with_capabilities"]), 
+            SetRemap("controller/robotic_status", [EnvironmentVariable('CARMA_INTR_NS', default_value=''), "/controller/robot_status"]),
+            SetRemap("controller/vehicle_cmd" , [EnvironmentVariable('CARMA_INTR_NS', default_value=''), "/controller/vehicle_cmd"]),
+            SetRemap("comms/outbound_binary_msg",[EnvironmentVariable('CARMA_INTR_NS', default_value=''), "/comms/outbound_binary_msg"]), 
+            SetRemap("comms/inbound_binary_msg", [EnvironmentVariable('CARMA_INTR_NS', default_value=''), "/comms/inbound_binary_msg"]),  
+            SetRemap("can/engine_speed", [EnvironmentVariable('CARMA_INTR_NS', default_value=''), "/can/engine_speed"]),
+            SetRemap("can/speed", [EnvironmentVariable('CARMA_INTR_NS', default_value=''), "/can/speed"]),
+            SetRemap("can/acc_engaged", [EnvironmentVariable('CARMA_INTR_NS', default_value=''), "/can/acc_engaged"]),
+            IncludeLaunchDescription(
+                FrontendLaunchDescriptionSource(
```
[the remark this answers is not in the document]
Comment (msmcconnell) [reply carma/launch/ui.launch.py]: done

File: platooning_strategic_IHP/src/platoon_strategic_ihp_node.cpp [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    if (worker_)
+      worker_->plan_maneuver_cb(*req, *resp);
+  }
+
+  bool Node::get_availability() {
+    return true;
+  }
+
+  std::string Node::get_version_id() {
+    return "v4.0";
```
Comment (adev4a) [inline platooning_strategic_IHP/src/platoon_strategic_ihp_node.cpp:233]: Should this be a configurable parameter?

Comment (msmcconnell) [reply platooning_strategic_IHP/src/platoon_strategic_ihp_node.cpp:233]: I don't think so, version numbers should always be impossible for users to change. If it was a configuration parameter, I could make no code changes, but pretend to run an older or newer version.

**Review by adev4a — APPROVED** [review]

**Pull request merged**